fdbrpc/include/fdbrpc/LoadBalance.actor.h

```diff
@@ -483,16 +498,23 @@ struct RequestData : NonCopyable {
 	                                      int requiredReplicas) {
 		if (model && (compareReplicas || FLOW_KNOBS->ENABLE_REPLICA_CONSISTENCY_CHECK_ON_READS)) {
 			ASSERT(requestStream != nullptr);
-			int requiredReplicaCount =
-			    compareReplicas ? requiredReplicas : FLOW_KNOBS->CONSISTENCY_CHECK_REQUIRED_REPLICAS;
-			return replicaComparison(request,
-			                         response,
-			                         requestStream->getEndpoint().token.first(),
-			                         alternatives,
-			                         channel,
-			                         requiredReplicaCount);
+			if (FLOW_KNOBS->ENABLE_REPLICA_CONSISTENCY_CHECK_ON_READS) {
+				return replicaComparison(request,
```

Comment:
I think you are changing the precedence here - the flow knob is getting a higher priority in the modified code. That is probably fine.

Reply:
Oh, I think we want to change it back. Let me do this.